app/test/consistent_apphash_test.go

```diff
@@ -171,3 +340,144 @@ func deterministicKeyRing(cdc codec.Codec) (keyring.Keyring, []types.PubKey) {
 	}
 	return kb, pubKeys
 }
+
+// processSdkMessages takes a list of sdk messages, forms transactions, signs them
+// and returns a list of raw transactions
+func processSdkMessages(signer *user.Signer, sdkMessages []sdk.Msg) ([][]byte, error) {
+	rawSdkTxs := make([][]byte, 0, len(sdkMessages))
+	for _, msg := range sdkMessages {
+		rawSdkTx, err := signer.CreateTx([]sdk.Msg{msg}, blobfactory.DefaultTxOpts()...)
+		if err != nil {
+			return nil, err
+		}
+
+		signerAddress := msg.GetSigners()[0]
+		signerAccount := signer.AccountByAddress(signerAddress)
+		err = signer.SetSequence(signerAccount.Name(), signerAccount.Sequence()+1)
+		if err != nil {
+			return nil, err
+		}
+
+		rawSdkTxs = append(rawSdkTxs, rawSdkTx)
+	}
+	return rawSdkTxs, nil
+}
+
+// executeTxs executes a set of transactions and returns the data hash and app hash
+func executeTxs(testApp *app.App, rawBlobTx []byte, rawSdkTxs [][]byte, validators []abci.Validator, lastCommitHash []byte) ([]byte, []byte, error) {
+	height := testApp.LastBlockHeight() + 1
+	chainID := testApp.GetChainID()
+
+	genesisTime := testutil.GenesisTime
+
+	// Prepare Proposal
+	resPrePareProposal := testApp.PrepareProposal(abci.RequestPrepareProposal{
+		BlockData: &tmproto.Data{
+			Txs: rawSdkTxs,
+		},
+		ChainId: chainID,
+		Height:  height,
+		// Dynamically increase time so the validator can be unjailed (1m duration)
+		Time: genesisTime.Add(time.Duration(height) * time.Minute),
+	})
+
+	dataHash := resPrePareProposal.BlockData.Hash
+
+	header := tmproto.Header{
+		Version:        version.Consensus{App: 1},
+		DataHash:       resPrePareProposal.BlockData.Hash,
+		ChainID:        chainID,
+		Time:           genesisTime.Add(time.Duration(height) * time.Minute),
+		Height:         height,
+		LastCommitHash: lastCommitHash,
+	}
+
+	// Process Proposal
+	resProcessProposal := testApp.ProcessProposal(abci.RequestProcessProposal{
+		BlockData: resPrePareProposal.BlockData,
+		Header:    header,
+	},
+	)
+	if abci.ResponseProcessProposal_ACCEPT != resProcessProposal.Result {
+		return nil, nil, fmt.Errorf("ProcessProposal failed: %v", resProcessProposal.Result)
+	}
+
+	// Begin block
+	validator3Signed := height == 2 // Validator 3 signs only the first block
+	testApp.BeginBlock(abci.RequestBeginBlock{
+		Header: header,
+		LastCommitInfo: abci.LastCommitInfo{
+			Votes: []abci.VoteInfo{
+				// In order to withdraw commission for this validator
+				{
+					Validator:       validators[0],
+					SignedLastBlock: true,
+				},
+				// In order to jail this validator
+				{
+					Validator:       validators[3],
+					SignedLastBlock: validator3Signed,
+				},
+			},
+		},
+	})
+
+	// Deliver SDK Txs
+	for i, rawSdkTx := range rawSdkTxs {
+		resp := testApp.DeliverTx(abci.RequestDeliverTx{Tx: rawSdkTx})
+		if resp.Code != uint32(0) {
```

Comment:
[nit] can use https://github.com/celestiaorg/celestia-core/blob/4d17212747d2fb5fd20723c6599a2c2990ddcb83/abci/types/result.go#L11 instead of `uint32(0)`